main/src/ca/uwaterloo/flix/language/phase/typer/TypeReduction.scala

```diff
@@ -293,13 +293,91 @@ object TypeReduction {
       case (Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Array, _), elmType1, _), rcVar1, _),
       Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Array, _), elmType2, _), rcVar2, _)) =>
         isSubtype(elmType1, elmType2)
+      // Arrow to Java function interface
+      // TODO INTEROP: generics support
+      // Pattern-matching Flix functions of the form: eff -> arg -> ret
+      // IntStream
+      case (Type.Apply(Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Arrow(n), _), eff, _), var_arg, _), var_ret, _), Type.Cst(TypeConstructor.Native(clazz), _))
+      if clazz == classOf[java.util.function.IntConsumer] =>
+        val targsVal = tpe1.typeArguments.map(purifyType)
+        targsVal match {
```

Comment:
and then here it should be `(var_arg, var_ret) match {case (Type.Cst(TypeConstructor.Int32, _), Type.Cst(TypeConstructor.Unit, _)) => ...` }

Reply:
oh and also not snake_case